backend/utils/bulk_operations.py

```
@@ -0,0 +1,289 @@
+    """
+    Update chemical status for all chemicals in a single operation
+    """
+    try:
+        now = datetime.utcnow()
+
+        # Update expired chemicals
+        expired_count = db.session.query(Chemical).filter(
+            and_(
+                Chemical.expiration_date < now,
+                Chemical.status != 'expired',
+                Chemical.is_archived == False
```

Ruff (0.11.9)

Avoid equality comparisons to `False`; use `if not Chemical.is_archived:` for false checks

Replace with `not Chemical.is_archived`

(E712)